src/v/cluster/partition_balancer_planner.h

```diff
@@ -67,6 +68,7 @@ class partition_balancer_planner {
         std::vector<model::node_id> all_nodes;
         absl::flat_hash_map<model::node_id, node_disk_space> node_disk_reports;
         absl::flat_hash_set<model::node_id> unavailable_nodes;
+        absl::flat_hash_set<model::node_id> nodes_in_maintenance;
```

Comment:
do we need a set of nodes being in maintenance mode, would counter be enough ?

Reply:
yeah, I can change it to a counter